test/pylib/minio_server.py

```diff
@@ -206,6 +206,13 @@ def _get_environs(self):
                 self.ENV_ACCESS_KEY,
                 self.ENV_SECRET_KEY]
 
+    def get_envs_settings(self):
```

Comment:
Without xdist there will be questions why we need this if this is slower than test.py. Even more, there are questions why not to add already use of this in test.py, since this logic is not used anywhere and there is no point to committing it in a half finished state. 
Changes in S3 and minio do not change the logic in any way. I'll ping the authors of these classes.

Reply:
I guess you also want to apply same changes to the `test/pylib/s3_proxy.py`
Other than that, lgtm

Reply:
Thanks. Modified as well.